pkg/common-controller/snapshot_controller.go

```diff
@@ -266,7 +270,12 @@ func (ctrl *csiSnapshotCommonController) checkandRemoveSnapshotFinalizersAndChec
 }
 
 // checkandAddSnapshotFinalizers checks and adds snapshot finailzers when needed
-func (ctrl *csiSnapshotCommonController) checkandAddSnapshotFinalizers(snapshot *crdv1.VolumeSnapshot, snapshotBound bool, deleteContent bool) {
+func (ctrl *csiSnapshotCommonController) checkandAddSnapshotFinalizers(snapshot *crdv1.VolumeSnapshot) error {
```

Comment:
Looks like the logic is to addSnapshotAsSourceFinalizer no matter whether there is PVC is currently using snapshot. I think it should be fine as long as we have correct logic to remove it when needed. But worth to put a note about it.

Reply:
Added note before NeedToAddSnapshotAsSourceFinalizer().